### DISCUSSION TITLE
Using on_close_requested callback for Window

### DISCUSSION THREAD

**Discussion by mrandal** [slint-ui/slint#6828] [Q&A] [answered]
How can I define the `on_close_requested()` callback in Rust? I am trying to define it as a normal callback, however I keep getting the error `error[E0599]: no method named 'on_close_requested' found for struct 'App' in the current scope`
App inherits window in the slint file

```rust
    let app = App::new().unwrap();
    app.on_close_requested(||{
         // code
         slint::CloseRequestResponse::HideWindow
    });
```
Here is the documentation - [https://snapshots.slint.dev/master/docs/rust/slint/struct.window#method.on_close_requested](https://snapshots.slint.dev/master/docs/rust/slint/struct.window#method.on_close_requested )

**Comment by Enyium** [comment, marked as answer]
You're trying to call `on_close_requested()` on the Rust struct the Slint compiler generates for you that implements the [`ComponentHandle`](https://snapshots.slint.dev/master/docs/rust/slint/trait.componenthandle) trait. But you first have to call that trait's `window()` function to retrieve the associated host window struct, called [`Window`](https://snapshots.slint.dev/master/docs/rust/slint/struct.window).

In #6027, I proposed to better name these things. `window()` and `Window`, e.g., could be named `host_window()`/`host()` and `HostWindow`. `ComponentHandle` could be named `WindowComponent`.
